packages/flutter_tools/gradle/src/test/kotlin/FlutterPluginUtilsTest.kt

```diff
@@ -0,0 +1,431 @@
+package com.flutter.gradle
+
+import com.android.builder.model.BuildType
+import io.mockk.every
+import io.mockk.mockk
+import io.mockk.verify
+import org.gradle.api.GradleException
+import org.gradle.api.Project
+import org.gradle.api.Task
+import org.gradle.api.UnknownTaskException
+import org.gradle.api.logging.Logger
+import org.junit.jupiter.api.assertThrows
+import org.junit.jupiter.api.io.TempDir
+import java.io.File
+import java.nio.file.Path
+import kotlin.test.Test
+import kotlin.test.assertEquals
+import kotlin.test.assertFalse
+import kotlin.test.assertTrue
+
+class FlutterPluginUtilsTest {
+    // toCamelCase
+    @Test
+    fun `toCamelCase converts a list of strings to camel case`() {
```

Comment:
As a style thing I do not like the quotes with spaces in the test name. But these tests are exactly what I was hoping to see when we did the conversion. Clean easy to follow easy to maintain tests.